libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -508,52 +500,29 @@ private List<ReturnCode> mapCisToReturnCodes(ContraIndicators contraIndicators)
                 .toList();
     }
 
-    private JsonNode getVcClaimNode(String credential, String node)
-            throws CredentialParseException {
-        try {
-            return OBJECT_MAPPER
-                    .readTree(SignedJWT.parse(credential).getPayload().toString())
-                    .path(VC_CLAIM)
-                    .path(node);
-        } catch (JsonProcessingException | ParseException e) {
-            throw new CredentialParseException(
-                    "Encountered a parsing error while attempting to parse VC store item: "
-                            + e.getMessage());
-        }
+    private List<NameParts> getNamePartsFromCredentialSubjectName(uk.gov.di.model.Name name) {
+        return name.getNameParts().stream()
+                .map(np -> new NameParts(np.getValue(), np.getType().toString()))
+                .toList();
     }
 
-    private <T> T getJsonProperty(JsonNode jsonNode, String propertyName, CollectionType valueType)
-            throws HttpResponseExceptionWithErrorBody {
-        JsonNode propertyNode = jsonNode.path(propertyName);
-        if (propertyNode.isMissingNode()) {
-            return OBJECT_MAPPER.convertValue(List.of(), valueType);
-        }
-        try {
-            return OBJECT_MAPPER.treeToValue(propertyNode, valueType);
-        } catch (JsonProcessingException e) {
-            LOGGER.error(LogHelper.buildErrorMessage("Failed to parse VC JWT", e));
-            throw new HttpResponseExceptionWithErrorBody(
-                    500, ErrorResponse.FAILED_TO_GENERATE_IDENTIY_CLAIM);
-        }
-    }
+    private IdentityClaim getIdentityClaim(VerifiableCredential vc) {
+        var credentialSubject =
+                ((IdentityCheckCredential) vc.getCredential()).getCredentialSubject();
```

Comment:
As below - prefer doing this in a checked cast:
```
if (vc.getCredential() instanceof IdentityCheckCredential credential) {
    var name = credential.getCredentialSubject().getName();
    ...
} else {
    // error - not an identity check credential!
}
```

However, I think you also want this to work for other types of credential? In fact, I think you can do something like:
```
if (vc.getCredential().getCredentialSubject() instanceof PersonWithIdentity person) {
    var name = person.getName();
    ...
} else {
    // No applicable subject in this credential
}
```
This should work for both `IdentityCheckCredential` and `IdentityAssertionCredential`